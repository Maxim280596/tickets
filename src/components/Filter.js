import React from 'react';
import {Button, Col, Row} from 'react-bootstrap';
import styled from 'styled-components';


const StyledButton = styled.button `
  width:50%;
  height:25%;
  margin-bottom: 20px;
  border-radius:10px;
  
`;


export  function Filter () {


  return(
    < >
    
    <StyledButton variant="primary" size="lg">Самый дешевый</StyledButton>
    <StyledButton variant="primary" size="lg">Самый быстрый</StyledButton>
    
    </>
  )
}