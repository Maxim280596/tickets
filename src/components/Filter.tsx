import React, { useState } from 'react';
import { useDispatch } from 'react-redux';

import { StyledButtonFast, StyledButtonPrice } from '../styled/StyledFilter';
import {
  filteringFast,
  filteringPrice,
} from '../redux/ducks/ticketsReduser';


const Filter = () => {
  const dispatch = useDispatch();
  const [price, setPrice] = useState(false);
  const [fast, setFast] = useState(false);
  const filterPrice = () => {
    dispatch(filteringPrice());
    setPrice(true);
    setFast(false);
  };

  const filterFast = () => {
    dispatch(filteringFast());
    setPrice(false);
    setFast(true);
  };

  return (
    <>
      <StyledButtonPrice onClick={filterPrice} active={price}>
        САМЫЙ ДЕШЕВЫЙ
      </StyledButtonPrice>
      <StyledButtonFast onClick={filterFast} active={fast}>
        САМЫЙ БЫСТРЫЙ
      </StyledButtonFast>
    </>
  );
};

export default Filter;
