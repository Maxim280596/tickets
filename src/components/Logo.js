import React from 'react';
import { Row, Container } from 'react-bootstrap';

import Logotype from '../logo.png';
import { StyledLogo, StyledLogoCol } from '../styled/StyledLogo';

const Logo = () => (
  <Container>
    <Row>
      <StyledLogoCol>
        <StyledLogo src={Logotype} alt="Logo" />
      </StyledLogoCol>
    </Row>
  </Container>
);

export default Logo;
