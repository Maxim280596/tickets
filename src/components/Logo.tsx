import Logotype from '../logo.png';
import { Row, Container } from 'react-bootstrap';

import { StyledLogo, StyledLogoCol } from '../styled/StyledLogo';

export const Logo = () => (
  <Container>
    <Row>
      <StyledLogoCol>
        <StyledLogo src={Logotype} alt="Logo" />
      </StyledLogoCol>
    </Row>
  </Container>
);

export default Logo;
