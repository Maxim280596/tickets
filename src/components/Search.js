import React, { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { BrowserRouter as Router, useHistory } from 'react-router-dom';

import {
  filteringOneStop,
  filteringAllStop,
  filteringNoStop,
  filteringThreeStop,
  filteringTwoStop,
} from '../src/redux/ducks/ticketsReducer';
import { StyledForm } from '../styled/StyledSearch';
import { SearchInput } from './SearchInput';

const Search = () => {
  const dispatch = useDispatch();

  const [state, setState] = useState({
    all: false,
    no: false,
    one: false,
    two: false,
    three: false,
  });

  let history = useHistory();

  const handleChange = (e) => {
    history.push(`/${e.target.name}`);
    setState({
      [e.target.name]: e.target.checked,
    });

    if (e.target.name === 'one') {
      dispatch(filteringOneStop());
    }
    if (e.target.name === 'two') {
      dispatch(filteringTwoStop());
    }
    if (e.target.name === 'three') {
      dispatch(filteringThreeStop());
    }
    if (e.target.name === 'all') {
      dispatch(filteringAllStop());
    }
    if (e.target.name === 'no') {
      dispatch(filteringNoStop());
    }
  };

  useEffect(() => {
    const { pathname } = history.location;
    setState({
      [pathname.substring(1)]: true,
    });
  }, []);

  return (
    <Router>
      <StyledForm>
        <div>
          <h5>КОЛИЧЕСТВО ПЕРЕСАДОК</h5>
          <SearchInput
            name="all"
            onChange={handleChange}
            id="all"
            inline
            label="Все"
            type="checkbox"
            checked={state.all}
          />
          <SearchInput
            name="no"
            onChange={handleChange}
            id="no"
            inline
            label="Без пересадок"
            type="checkbox"
            checked={state.no}
          />
          <SearchInput
            name="one"
            onChange={handleChange}
            id="one"
            inline
            label="1 пересадка"
            type="checkbox"
            checked={state.one}
          />
          <SearchInput
            name="two"
            onChange={handleChange}
            id="two"
            inline
            label="2 пересадки"
            type="checkbox"
            checked={state.two}
          />
          <SearchInput
            name="three"
            onChange={handleChange}
            id="three"
            inline
            label="3 пересадки"
            type="checkbox"
            checked={state.three}
          />
        </div>
      </StyledForm>
    </Router>
  );
};

export default Search;
