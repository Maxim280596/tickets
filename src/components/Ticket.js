import React from 'react';
import { Row, Col } from 'react-bootstrap';

import {
  StyledDiv,
  StyledRow,
  StyledPrice,
  StyledImg,
  StyledTitle,
  StyledValue,
} from '../styled/StyledTicket';

export const Ticket = ({
  price,
  origin,
  destination,
  originBack,
  destinationBack,
  stops,
  stopsBack,
  stopsCity,
  stopsCityBack,
  departureDate,
  duration,
  departureDateBack,
  durationBack
}) => {


  const getTimeToString = (date, duration) => {
    let dateStart = new Date(date);
    let dateEnd = new Date(dateStart.getTime() + duration * 60 * 1000);
    return dateStart.toTimeString().slice(0, 5) + " - " + dateEnd.toTimeString().slice(0, 5);
};

const getDurationToString = (duration) => {
  return Math.floor(duration / 60).toString() + "ч " + (duration % 60) + "м";
};

  return (
    <StyledRow>
      <Row>
        <Col lg={4} sm={4} xs={4}>
          <StyledPrice>{price} р.</StyledPrice>
        </Col>
        <Col
          lg={{ span: 4, offset: 4 }}
          sm={{ span: 4, offset: 4 }}
          xs={{ span: 5, offset: 3 }}
        >
          <div>
            <StyledImg
              src="https://upload.wikimedia.org/wikipedia/commons/c/c2/S7_new_logo.svg"
              alt="s7 airlines logo"
            />
          </div>
        </Col>

        <Col lg={4} sm={4} xs={4}>
          <StyledDiv>
            <StyledTitle>
              {origin}-{destination}
            </StyledTitle>
            <StyledValue>{getTimeToString(departureDate,duration)}</StyledValue>
          </StyledDiv>
          <StyledDiv>
            <StyledTitle>
              {originBack}-{destinationBack}
            </StyledTitle>
            <StyledValue>{getTimeToString(departureDateBack,durationBack)}</StyledValue>
          </StyledDiv>
        </Col>
        <Col lg={4} sm={4} xs={4}>
          <StyledDiv>
            <StyledTitle>В пути</StyledTitle>
            <StyledValue>{getDurationToString(duration)}</StyledValue>
          </StyledDiv>
          <StyledDiv>
            <StyledTitle>В пути</StyledTitle>
            <StyledValue>{getDurationToString(durationBack)}</StyledValue>
          </StyledDiv>
        </Col>
        <Col lg={4} sm={4} xs={4}>
          <StyledDiv>
            <StyledTitle>{stops} пересадки</StyledTitle>
            <StyledValue>{stopsCity}</StyledValue>
          </StyledDiv>
          <StyledDiv>
            <StyledTitle>{stopsBack} пересадки</StyledTitle>
            <StyledValue>{stopsCityBack}</StyledValue>
          </StyledDiv>
        </Col>
      </Row>
    </StyledRow>
  );
};
