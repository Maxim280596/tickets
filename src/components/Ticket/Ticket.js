import React from "react";
import { Row, Col } from "react-bootstrap";

import {
  StyledDiv,
  StyledRow,
  StyledPrice,
  StyledImg,
  StyledTitle,
  StyledValue,
} from "./StyledTicket";

export function Ticket() {
  return (
    <StyledRow>
      <Row>
        <Col lg={4} sm={4} xs={4}>
          <StyledPrice>1780 P</StyledPrice>
        </Col>
        <Col
          lg={{ span: 4, offset: 4 }}
          sm={{ span: 4, offset: 4 }}
          xs={{ span: 5, offset: 3 }}
        >
          <div>
            <StyledImg
              src="https://upload.wikimedia.org/wikipedia/commons/c/c2/S7_new_logo.svg"
              alt="s7 airlines logo"
            />
          </div>
        </Col>

        <Col lg={4} sm={4} xs={4}>
          <StyledDiv>
            <StyledTitle>HWT-KWT</StyledTitle>
            <StyledValue>10:45-08:00</StyledValue>
          </StyledDiv>
          <StyledDiv>
            <StyledTitle>HWT-KWT</StyledTitle>
            <StyledValue>10:45-08:00</StyledValue>
          </StyledDiv>
        </Col>
        <Col lg={4} sm={4} xs={4}>
          <StyledDiv>
            <StyledTitle>В пути</StyledTitle>
            <StyledValue>21ч 15мин</StyledValue>
          </StyledDiv>
          <StyledDiv>
            <StyledTitle>В пути</StyledTitle>
            <StyledValue>21ч 15мин</StyledValue>
          </StyledDiv>
        </Col>
        <Col lg={4} sm={4} xs={4}>
          <StyledDiv>
            <StyledTitle>2 пересадки</StyledTitle>
            <StyledValue>HKB, JNB</StyledValue>
          </StyledDiv>
          <StyledDiv>
            <StyledTitle>2 пересадки</StyledTitle>
            <StyledValue>HKB, JNB</StyledValue>
          </StyledDiv>
        </Col>
      </Row>
    </StyledRow>
  );
}
