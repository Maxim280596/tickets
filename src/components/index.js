import { Search } from "./Search/Search";
import { Filter } from "./Filter/Filter";
import { TicketsList } from "./TicketList/TicketsList";
import { Logo } from "./Logo/Logo";

export {Logo,TicketsList,Filter,Search};