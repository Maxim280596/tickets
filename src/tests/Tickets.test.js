import React from 'react';
import {render} from 'enzyme';
import toJson from 'enzyme-to-json';

import {Ticket} from '../components/Ticket';

describe('Testing <Ticket/>', () => {
  it('Logo have rendered correctly', ()=>{
    const ticket = render(<Ticket price= '100'
      origin='MOW'
      destination='HKT'
      originBack='HKT'
      destinationBack='MOW'
      stops='3'
      stopsBack='2'
      stopsCity='KV,LG,HKT'
      stopsCityBack='KV,LG,HKT'
      departureDate='12342'
      duration='121323'
      departureDateBack='232333'
      durationBack='232'/>);
    expect(toJson(ticket)).toMatchSnapshot();
  })
})