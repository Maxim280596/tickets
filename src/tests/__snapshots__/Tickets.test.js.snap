// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing rendering <Ticket/> should rendering Ticket correctly without props 1`] = `
<Ticket>
  <styled.div>
    <div
      className="sc-bdfBwQ kbqjdu"
    >
      <Row
        noGutters={false}
      >
        <div
          className="row"
        >
          <Col
            lg={4}
            sm={5}
            xs={5}
          >
            <div
              className="col-lg-4 col-sm-5 col-5"
            >
              <Styled(Col)>
                <Col
                  className="sc-gsTCUz jzgiUW"
                >
                  <div
                    className="sc-gsTCUz jzgiUW col"
                  >
                     р.
                  </div>
                </Col>
              </Styled(Col)>
            </div>
          </Col>
          <Col
            lg={
              Object {
                "offset": 4,
                "span": 4,
              }
            }
            sm={
              Object {
                "offset": 3,
                "span": 4,
              }
            }
            xs={
              Object {
                "offset": 2,
                "span": 5,
              }
            }
          >
            <div
              className="col-lg-4 col-sm-4 col-5 offset-lg-4 offset-sm-3 offset-2"
            >
              <div>
                <styled.img
                  alt="s7 airlines logo"
                  src="https://upload.wikimedia.org/wikipedia/commons/c/c2/S7_new_logo.svg"
                >
                  <img
                    alt="s7 airlines logo"
                    className="sc-dlfnbm iBFTdI"
                    src="https://upload.wikimedia.org/wikipedia/commons/c/c2/S7_new_logo.svg"
                  />
                </styled.img>
              </div>
            </div>
          </Col>
          <Col
            lg={4}
            sm={4}
            xs={4}
          >
            <div
              className="col-lg-4 col-sm-4 col-4"
            >
              <styled.div>
                <div
                  className="sc-hKgILt hNEWQR"
                >
                  <styled.span>
                    <span
                      className="sc-eCssSg jJBKym"
                    >
                      -
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-jSgupP lcodnf"
                    >
                      Inval - Inval
                    </span>
                  </styled.span>
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-hKgILt hNEWQR"
                >
                  <styled.span>
                    <span
                      className="sc-eCssSg jJBKym"
                    >
                      -
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-jSgupP lcodnf"
                    >
                      Inval - Inval
                    </span>
                  </styled.span>
                </div>
              </styled.div>
            </div>
          </Col>
          <Col
            lg={4}
            sm={4}
            xs={4}
          >
            <div
              className="col-lg-4 col-sm-4 col-4"
            >
              <styled.div>
                <div
                  className="sc-hKgILt hNEWQR"
                >
                  <styled.span>
                    <span
                      className="sc-eCssSg jJBKym"
                    >
                      В пути
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-jSgupP lcodnf"
                    >
                      NaNч NaNм
                    </span>
                  </styled.span>
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-hKgILt hNEWQR"
                >
                  <styled.span>
                    <span
                      className="sc-eCssSg jJBKym"
                    >
                      В пути
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-jSgupP lcodnf"
                    >
                      NaNч NaNм
                    </span>
                  </styled.span>
                </div>
              </styled.div>
            </div>
          </Col>
          <Col
            lg={4}
            sm={4}
            xs={4}
          >
            <div
              className="col-lg-4 col-sm-4 col-4"
            >
              <styled.div>
                <div
                  className="sc-hKgILt hNEWQR"
                >
                  <styled.span>
                    <span
                      className="sc-eCssSg jJBKym"
                    >
                       пересадки
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-jSgupP lcodnf"
                    />
                  </styled.span>
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-hKgILt hNEWQR"
                >
                  <styled.span>
                    <span
                      className="sc-eCssSg jJBKym"
                    >
                       пересадки
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-jSgupP lcodnf"
                    />
                  </styled.span>
                </div>
              </styled.div>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </styled.div>
</Ticket>
`;

exports[`Testing rendering <Ticket/> should rendering Ticket with props 1`] = `
<Ticket
  departureDate="12342"
  departureDateBack="232333"
  destination="HKT"
  destinationBack="MOW"
  duration="121323"
  durationBack="232"
  origin="MOW"
  originBack="HKT"
  price="100"
  stops="3"
  stopsBack="2"
  stopsCity="KV,LG,HKT"
  stopsCityBack="KV,LG,HKT"
>
  <styled.div>
    <div
      className="sc-bdfBwQ kbqjdu"
    >
      <Row
        noGutters={false}
      >
        <div
          className="row"
        >
          <Col
            lg={4}
            sm={5}
            xs={5}
          >
            <div
              className="col-lg-4 col-sm-5 col-5"
            >
              <Styled(Col)>
                <Col
                  className="sc-gsTCUz jzgiUW"
                >
                  <div
                    className="sc-gsTCUz jzgiUW col"
                  >
                    100
                     р.
                  </div>
                </Col>
              </Styled(Col)>
            </div>
          </Col>
          <Col
            lg={
              Object {
                "offset": 4,
                "span": 4,
              }
            }
            sm={
              Object {
                "offset": 3,
                "span": 4,
              }
            }
            xs={
              Object {
                "offset": 2,
                "span": 5,
              }
            }
          >
            <div
              className="col-lg-4 col-sm-4 col-5 offset-lg-4 offset-sm-3 offset-2"
            >
              <div>
                <styled.img
                  alt="s7 airlines logo"
                  src="https://upload.wikimedia.org/wikipedia/commons/c/c2/S7_new_logo.svg"
                >
                  <img
                    alt="s7 airlines logo"
                    className="sc-dlfnbm iBFTdI"
                    src="https://upload.wikimedia.org/wikipedia/commons/c/c2/S7_new_logo.svg"
                  />
                </styled.img>
              </div>
            </div>
          </Col>
          <Col
            lg={4}
            sm={4}
            xs={4}
          >
            <div
              className="col-lg-4 col-sm-4 col-4"
            >
              <styled.div>
                <div
                  className="sc-hKgILt hNEWQR"
                >
                  <styled.span>
                    <span
                      className="sc-eCssSg jJBKym"
                    >
                      MOW
                      -
                      HKT
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-jSgupP lcodnf"
                    >
                      00:00 - 06:03
                    </span>
                  </styled.span>
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-hKgILt hNEWQR"
                >
                  <styled.span>
                    <span
                      className="sc-eCssSg jJBKym"
                    >
                      HKT
                      -
                      MOW
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-jSgupP lcodnf"
                    >
                      00:00 - 03:52
                    </span>
                  </styled.span>
                </div>
              </styled.div>
            </div>
          </Col>
          <Col
            lg={4}
            sm={4}
            xs={4}
          >
            <div
              className="col-lg-4 col-sm-4 col-4"
            >
              <styled.div>
                <div
                  className="sc-hKgILt hNEWQR"
                >
                  <styled.span>
                    <span
                      className="sc-eCssSg jJBKym"
                    >
                      В пути
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-jSgupP lcodnf"
                    >
                      2022ч 3м
                    </span>
                  </styled.span>
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-hKgILt hNEWQR"
                >
                  <styled.span>
                    <span
                      className="sc-eCssSg jJBKym"
                    >
                      В пути
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-jSgupP lcodnf"
                    >
                      3ч 52м
                    </span>
                  </styled.span>
                </div>
              </styled.div>
            </div>
          </Col>
          <Col
            lg={4}
            sm={4}
            xs={4}
          >
            <div
              className="col-lg-4 col-sm-4 col-4"
            >
              <styled.div>
                <div
                  className="sc-hKgILt hNEWQR"
                >
                  <styled.span>
                    <span
                      className="sc-eCssSg jJBKym"
                    >
                      3
                       пересадки
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-jSgupP lcodnf"
                    >
                      KV,LG,HKT
                    </span>
                  </styled.span>
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-hKgILt hNEWQR"
                >
                  <styled.span>
                    <span
                      className="sc-eCssSg jJBKym"
                    >
                      2
                       пересадки
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-jSgupP lcodnf"
                    >
                      KV,LG,HKT
                    </span>
                  </styled.span>
                </div>
              </styled.div>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </styled.div>
</Ticket>
`;
